src/App.jsx

```diff
@@ -1,28 +1,87 @@
 import React from 'react';
 import './App.scss';
 import './styles/general.scss';
+import { getTodos } from './api';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
 
 class App extends React.Component {
   state = {
     todos: [],
+    todosForRender: [],
     selectedUserId: 0,
   };
 
+  componentDidMount() {
+    getTodos()
+      .then(todos => this.setState({
+        todos,
+        todosForRender: todos,
+      }));
+  }
+
+  selectUser = (userId) => {
+    this.setState({
+      selectedUserId: userId,
+    });
+  }
+
+  clearUser = () => {
+    this.setState({
+      selectedUserId: 0,
+    });
+  }
+
+  handleChange = (query) => {
+    this.setState(state => ({
+      todosForRender: [...state.todos]
```

Comment:
no need to copy it, filter doesn't mutate the array

Reply:
fixed all the comments, and these buttons select all the todos of the current user  https://prnt.sc/ufmyez